ui/src/__tests__/pages/AllStories.test.ts

```diff
@@ -73,15 +75,44 @@ describe("allstories table", () => {
       };
       expect(INIT_STATE.visibleTableState.length).toBe(0);
       expect(INIT_STATE.origTableData.length).toBe(0);
-      expect(INIT_STATE.origTableData.length).toBe(0);
+      expect(INIT_STATE.tableData.length).toBe(0);
+      expect(INIT_STATE.tabValue).toBe(0);
       const newState = allStoriesReducer(INIT_STATE, action);
       expect(newState.visibleTableState.length).toBe(2);
       expect(newState.tableData.length).toBe(2);
       expect(newState.origTableData.length).toBe(2);
     });
 
+    it("should initialize tags and filter states after Tags API", () => {
+      const tagRows = ["TAG1", "TAG2"];
+      const action: Action = {
+        type: "INITIALIZE_AFTER_TAGS_API",
+        rows: tagRows,
+      };
+      expect(INIT_STATE.tags.length).toBe(0);
+      expect(INIT_STATE.filterState.tags).toEqual({});
+      expect(INIT_STATE.filterState.visibility).toEqual({
```

Comment:
hmm technically you're not expecting `filterState` to change in this test so you don't need the two expect() here.

Reply:
I am expecting it to change, becuase once tags are fetched, I am creating a filterState which depends on the tags that were fetched by the API